src/redux/reducers/local.js

```diff
@@ -0,0 +1,52 @@
+import actions from "./../constants/actions";
+
+export const initialState = {
+  likes: [],
+  ids: [],
+}
+
+export default (state = initialState, action) => {
+  const { type, payload } = action;
+
+  switch (type) {
+    case actions.LOCALSTORAGE_DATA: {
+      const { data } = payload;
+
+      let newState = { ...state };
+      const keys = Object.keys(data);
+
+      if (keys.length > 0) {
+        newState = { ...data };
+      }
+
+      return newState;
+    }
+    case actions.LOCALSTORAGE_MODIFY: {
```

Comment:
I would suggest you to invert the data flow regarding the `store` and `localStorage`. You should first store the favorites in the Store and have a subscription/saga/middleware hearing the proper actions to just dump the state to localStorage when needed.

Reply:
Actually if you check after LOCALSTORAGE_MODIFY redux saga runs localModifyStorage
whichs inject that in the window.localStorage 

check inside sagas folder: local.js
localModifyStorage generator